VideoWeb/VideoWeb/Controllers/ParticipantsController.cs

```diff
@@ -247,5 +248,58 @@ public async Task<IActionResult> GetParticipantsByConferenceIdAsync(Guid confere
             }
         }
 
+        [HttpGet("{conferenceId}/currentparticipant")]
+        [SwaggerOperation(OperationId = "GetCurrentParticipant")]
+        [ProducesResponseType(typeof(CurrentUserOrParticipantResponse), (int)HttpStatusCode.OK)]
+        [ProducesResponseType((int)HttpStatusCode.NotFound)]
+        public async Task<IActionResult> GetCurrentParticipantAsync(Guid conferenceId)
+        {
+
+            var participantsRoles = new List<Role>
+            {
+                Role.Judge,
+                Role.Individual,
+                Role.Representative,
+                Role.JudicialOfficeHolder
+            };
+
+            try
+            {
+                var claimsPrincipalToUserProfileResponseMapper = _mapperFactory.Get<ClaimsPrincipal, UserProfileResponse>();
+                var profile = claimsPrincipalToUserProfileResponseMapper.Map(User);
+                var response = new CurrentUserOrParticipantResponse
+                {
+                    AdminUsername = User.Identity.Name,
+                    DisplayName = "Admin",
+                    Role = Role.VideoHearingsOfficer
+                };
+
+                if (participantsRoles.Any(roleParticipant => roleParticipant == profile.Role))
+                {
+                    var conference = await _conferenceCache.GetOrAddConferenceAsync(conferenceId,
+                                   () => _videoApiClient.GetConferenceDetailsByIdAsync(conferenceId));
+                    if (conference != null)
+                    {
+                        var participant = conference.Participants
+                        .Single(x => x.Username.Equals(profile.Username, StringComparison.CurrentCultureIgnoreCase));
+
+                        response = new CurrentUserOrParticipantResponse
```

Comment:
Not sure im a huge fan of having "Or" in the name.. could we rename or split?

Reply:
Difficult task! Could it be: LoggedParticipantResponse? Is it Better?